File: 1-two-sum/two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    // This will hold the number : index pair
    const dict = new Map<number, number>();

    for (let i = 0; i < nums.length; i++) {
        const complement = target - nums[i];

        if (dict.has(complement)) {
            return [i, dict.get(complement)!];
        }

        // Put the number as key, and value as the index
        dict.set(nums[i], i);
    }

    return [];
}

File: 1-two-sum/two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        // Value : Index pairs 
        var freqDict = new Dictionary<int , int>();
        int[] output = new int[2];
        for(int i =0; i < nums.Length; i++){
            int compliment = target - nums[i];
            if(freqDict.ContainsKey(compliment)){
                output[0] = i;
                output[1] = freqDict[compliment];
                return output;
            }
            freqDict[nums[i]] = i;
        }
        return output;
    }
}